### PR TITLE
Change DoEliminateDeadLoop Check

### PR DESCRIPTION
**Pull request by SamirDroubi**
* Change the check from `lo >= hi == True` into `lo >= hi`. The former
    condition wouldn't converge in the z3 solver for the added tests.
* Refactored the Check_ExprBound to directly use strings instead of an
enum. This is generally how we have represented those operations within
the compiler and it is less code to write when calling the check.
* Deleted a duplicated DoEliminateIfDeadBranch function.

### CODE REVIEW

**Review by yamaguchi1024 — APPROVED** [review]
> The former condition wouldn't converge in the z3 solver for the added tests.

omg really? So cryptic..

File: src/exo/LoopIR_scheduling.py [hunk 1]
```diff
@@ -3808,7 +3778,7 @@ def guard_wrapper(body):
         # the forwarded version. However, in all the current callees, the
         # statement would have been just constructed and if you try to forward
         # you get an error.
-        ir, fwd_wrap = ctxt_stmt_c.parent().body()._wrap(guard_wrapper, "body")
+        ir, fwd_wrap = ctxt_stmt_c.as_block()._wrap(guard_wrapper, "body")
```
Comment (yamaguchi1024) [inline src/exo/LoopIR_scheduling.py:3781]: I think this change is from another PR

Comment (SamirDroubi) [reply src/exo/LoopIR_scheduling.py:3781]: Yeah, when I merge them inorder. This will go away from this PR

Comment (yamaguchi1024) [reply src/exo/LoopIR_scheduling.py:3781]: sg

**Pull request merged**